### PR TITLE
Add Image support 2

### PR DESCRIPTION
**Pull request by novakov-alexey**
I have created new PR, previous #271 was created from old fork state. 
This issue is still relevant: https://github.com/alexarchambault/plotly-scala/pull/271#issuecomment-798767433

### CODE REVIEW

**Comment by novakov-alexey** [comment]
I didn't get how to do documentation test. It seems Plotly documentation has moved to another repository. See here: https://github.com/plotly/documentation. 

Also, I can't find `Image` documentation containing some JavaScript example, only such reference page https://github.com/plotly/graphing-library-docs/blob/6646a9bcb68b10acf5f4708d58d7c16db11383ee/_posts/reference_pages/javascript/2020-07-20-image.html

**Comment by alexarchambault** [comment]
It seems there are some image examples in the plotly documentation, under `plotly-documentation/_posts/plotly_js/fundamentals/images`. These can be enabled by adding `"fundamentals/images"` around [here](https://github.com/alexarchambault/plotly-scala/blob/102f86b1740ffaa44fd338d89958f78751c9c473/tests/src/test/scala/plotly/doc/DocumentationTests.scala#L250).

But it seems images are added in the layout in these examples. Is it what you expected? The `Image` class should be moved under the `plotly.layout` namespace if that's the case.

**Comment by novakov-alexey** [comment]
> It seems there are some image examples in the plotly documentation, under `plotly-documentation/_posts/plotly_js/fundamentals/images`. These can be enabled by adding `"fundamentals/images"` around [here](https://github.com/alexarchambault/plotly-scala/blob/102f86b1740ffaa44fd338d89958f78751c9c473/tests/src/test/scala/plotly/doc/DocumentationTests.scala#L250).
> 
> But it seems images are added in the layout in these examples. Is it what you expected? The `Image` class should be moved under the `plotly.layout` namespace if that's the case.

Just looked at layout/images as well. Looks like these are to display images from PNG using URL to that image. This is a bit different use-case than Traces/Image that I have added in this PR. The last one allows to draw a 2D image from pixels array on the plot. Think of MNIST dataset visualization where you have grey-scaled or RGB images as 2D matrix of integers (1 - 3). We can create such pixels array programmatically and draw any rasters. 

As for documentation path, I would expect some JS examples here: https://github.com/plotly/graphing-library-docs/tree/master/_posts/plotly_js/basic , but there is no basic/image :-(

The only piece of documentation I have found on the subject is here: https://plotly.com/javascript/reference/image/
Not sure if it helps us.

File: core/shared/src/main/scala/plotly/element/ColorModel.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package plotly.element
+
+sealed abstract class ColorModel(val label: String) extends Product with Serializable
```
[the remark this answers is not in the document]
Comment (novakov-alexey) [reply core/shared/src/main/scala/plotly/element/ColorModel.scala:3]: Thanks. Done

File: core/shared/src/main/scala/plotly/element/Alignment.scala [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package plotly.element
+
+sealed abstract class Alignment(val label: String) extends Product with Serializable
```
[the remark this answers is not in the document]
Comment (novakov-alexey) [reply core/shared/src/main/scala/plotly/element/Alignment.scala:3]: Done

**Comment by novakov-alexey** [comment]
what else needs to be done to merge this PR?

**Comment by alexarchambault** [comment]
Sorry, meant to come back at this earlier...

Just adding a simple non regression tests would be fine. Rather than relying on snippets from the plotly documentation, we could just write our own. A test could be added at the end of [`DocumentationTests`](https://github.com/alexarchambault/plotly-scala/blob/102f86b1740ffaa44fd338d89958f78751c9c473/tests/src/test/scala/plotly/doc/DocumentationTests.scala), calling [`plotlyDemoElements`](https://github.com/alexarchambault/plotly-scala/blob/102f86b1740ffaa44fd338d89958f78751c9c473/tests/src/test/scala/plotly/doc/DocumentationTests.scala#L159) with a custom snippet, creating an image in JS, with some of the fields of `plotly.Image`. Just so that image support doesn't regress in the future.

**Comment by novakov-alexey** [comment]
@alexarchambault just added such test. Please have a look.

**Review by alexarchambault — APPROVED** [review]
Thanks!

**Comment by alexarchambault** [comment]
@novakov-alexey I squashed some commits manually, and pushed that directly to `master`. This PR changes are now in `master`, closing it.

**Comment by novakov-alexey** [comment]
@alexarchambault perfect, thanks!

**Pull request closed without merging**